transports/http/endpoints/api/access/endpoints.go

```diff
@@ -59,7 +59,14 @@ func (h *handler) signIn(c *gin.Context) {
 		return
 	}
 
-	signInUser, err := h.service.SignInUser(reqUser.Email, reqUser.Password)
+	exchangeRate, err := h.services.RatesService.GetExchangeRate()
+	if err != nil {
+		h.log.Error().Msgf("Exchange rate not found: %s", err)
+		c.JSON(http.StatusBadRequest, "An error occurred while getting exchange rate")
+		return
+	}
+
+	signInUser, err := h.services.UsersService.SignInUser(reqUser.Email, reqUser.Password, exchangeRate.Rate)
```

Comment:
I would rather make a RateService dependency of UserService (I mean, add it to UserService when creating it).
Because it looks a little bit weird, that when you read line with call to SignInUser
> Sing in user with email and password and exchange rate

Reply:
Done, in this situation I thought that we do not want to have dependency in one service from another service